### PR TITLE
Add ensures to axiom patterns

### PR DESCRIPTION
**Pull request by traiansf**
###### Reviewer checklist

- [ ] Test coverage: `stack test --coverage`
- [ ] Public API documentation: `stack haddock`
- [ ] Style conformance: `stylish-haskell`

---

### CODE REVIEW

**Comment by traiansf** [comment]
Jankins: test this please

**Comment by traiansf** [comment]
Jenkins: test this please

**Comment by ttuegel** [comment]
@traiansf You requested my review on this, but it is still marked "draft"; is it ready to go?

**Comment by vladciobanu** [comment]
@ttuegel Traian confirmed it is ready for review. I pushed the button for him :)

File: kore/test/Test/Kore/Step/BaseStep.hs [hunk 1]
```diff
@@ -723,11 +729,47 @@ test_baseStep =
...
                     , attributes = def
                     }
                 )
         assertEqualWithExplanation "" expect actual
 
+    -- x => x ensures g(x)=f(x)
+    -- vs
+    -- a
+    -- Expected: a and g(a)=f(a)
+    , testCase "Conjoins axiom pre-condition" $ do
```
Comment (ttuegel) [inline kore/test/Test/Kore/Step/BaseStep.hs]:
```suggestion
    , testCase "Conjoins axiom post-condition" $ do
```

**Pull request merged**